pytorch_lightning/trainer/data_loading.py

```diff
@@ -155,7 +155,7 @@ def _get_distributed_sampler(self, dataloader, train):
             assert self.distributed_backend is not None
             kwargs = dict(num_replicas=world_size[self.distributed_backend], rank=self.global_rank)
 
-        kwargs['shuffle'] = train
+        kwargs['shuffle'] = train and not self.overfit_batches
```

Comment:
should we have a test for this?

Reply:
this bug only affects ddp, but I don't know how to test this, since `distributed_backend="ddp"` does not work in CI.

Reply:
@SkafteNicki I tried to find a way but the problem is that this bug affects only ddp, and don't know how to test this due to the multiprocessing.